src/applications/check-in/components/pages/ConfirmablePage/index.jsx

```diff
@@ -19,9 +22,18 @@ const ConfirmablePage = ({
   noAction = () => {},
   withBackButton = false,
   pageType,
+  router,
 }) => {
   const { t } = useTranslation();
 
+  const { jumpToPage } = useFormRouting(router);
+  const { getCheckinComplete } = useSessionStorage(false);
```

Comment:
So I may have opened a can of worms with the request to move the routing here. Currently, this breaks pre-check-in when this session value is true. So what you would need to do is to select the APP from the reducer and pass that here instead of a `false`. So it would be `useSessionStorage(app === APP_NAMES.PRE_CHECK_IN)` That way it would return false for preCheckIn always since it would use those sessions storage keys. The other thing is that we should now pass the router to this component from the preCheckIn side.

Or I would also be fine going back to the first iteration where you were doing this in three places if you feel like this overhead outweighs the DRYness of moving the action here..... Sorry

Reply:
That makes sense and I don't think it's too much overhead. I pushed the suggested fixes.